adapter/adapter.go

```diff
@@ -0,0 +1,25 @@
+package adapter
+
+import (
+	"errors"
+	"github.com/c3pm-labs/c3pm/adapter/defaultadapter"
+	"github.com/c3pm-labs/c3pm/config"
+)
+
+type Adapter interface {
+	// Build builds the targets
+	Build(pc *config.ProjectConfig) error
+	// Targets return the paths of the targets built by the Build function
+	Targets(pc *config.ProjectConfig) (targets []string, err error)
+}
+
+func FromPC(pc *config.ProjectConfig) (Adapter, error) {
```

Comment:
Here I would maybe use the full name `FromProjectConfig`, seems a bit more clear